### ISSUE TITLE
Resource routes without browser exports cause errors in browser (Vite - Nightly)

### ISSUE THREAD

**Issue by mikearnaldi** [remix-run/remix#7795]
### What version of Remix are you using?

nightly

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

I am using the unstable vite plugin while testing an integration with effect.

Clone https://github.com/mikearnaldi/vite-remix-effect/tree/chore/bug-bundling

Run `pnpm i && pnpm build`

The issue is that in the route https://github.com/mikearnaldi/vite-remix-effect/blob/chore/bug-bundling/app/routes/bugged.tsx the loader code is included in the client bundle.

```ts
import { redirect } from "@remix-run/node";
import { Effect } from "effect";
import { effectLoader } from "~/services/Runtime";

export const loader = effectLoader(Effect.sync(() => redirect("/")));
```

When adding a component as a default export the issue disappears. My feeling is that the compiler fails to eliminate `action` and `loader` when there is no default export

### Expected Behavior

The following is the output with the route changed to:

```ts
import { redirect } from "@remix-run/node";
import { Effect } from "effect";
import { effectLoader } from "~/services/Runtime";

export const loader = effectLoader(Effect.sync(() => redirect("/")));

export function Bugged() {
  return <div></div>;
}
```

As you can see the code of `bugged` is significantly smaller and doesn't include loader code

```
vite v4.5.0 building for production...

  ⚠️  Remix support for Vite is unstable
     and not recommended for production

✓ 50 modules transformed.
public/build/manifest.json                      1.06 kB │ gzip:  0.26 kB
public/build/assets/bugged-a7e55202.js          0.10 kB │ gzip:  0.11 kB
public/build/assets/root-9fa1810b.js            1.40 kB │ gzip:  0.82 kB
public/build/assets/_index-7668716d.js          1.48 kB │ gzip:  0.67 kB
public/build/assets/jsx-runtime-0fbd76cf.js     7.81 kB │ gzip:  3.00 kB
public/build/assets/components-64e00afe.js     78.34 kB │ gzip: 24.97 kB
public/build/assets/entry.client-cb509d92.js  143.08 kB │ gzip: 46.17 kB
✓ built in 1.34s
vite v4.5.0 building SSR bundle for production...

  ⚠️  Remix support for Vite is unstable
     and not recommended for production

✓ 16 modules transformed.
build/assets/00001_create_todos-0f9ff181.js   0.35 kB
build/assets/00002_add_todos-e920757f.js      0.36 kB
build/index.js                               10.07 kB
✓ built in 364ms
```

### Actual Behavior

```
✓ 863 modules transformed.
public/build/manifest.json                            1.81 kB │ gzip:   0.37 kB
public/build/assets/bugged-d7929f89.js                0.09 kB │ gzip:   0.10 kB
public/build/assets/00001_create_todos-d10ac45c.js    0.34 kB │ gzip:   0.27 kB
public/build/assets/00002_add_todos-f25914cc.js       0.35 kB │ gzip:   0.23 kB
public/build/assets/root-400441ed.js                  1.39 kB │ gzip:   0.82 kB
public/build/assets/_index-d096f098.js                1.47 kB │ gzip:   0.67 kB
public/build/assets/components-2967d421.js           44.44 kB │ gzip:  14.16 kB
public/build/assets/router-6a14ae3e.js               47.71 kB │ gzip:  16.00 kB
public/build/assets/entry.client-214ef79c.js        143.08 kB │ gzip:  46.17 kB
public/build/assets/bugged-29e8fb7c.js              467.75 kB │ gzip: 132.26 kB
✓ built in 4.25s
vite v4.5.0 building SSR bundle for production...

  ⚠️  Remix support for Vite is unstable
     and not recommended for production

✓ 16 modules transformed.
build/assets/00001_create_todos-0f9ff181.js   0.35 kB
build/assets/00002_add_todos-e920757f.js      0.36 kB
build/index.js                               10.04 kB
✓ built in 366ms
```

**Comment by pcattori** [comment]
@mikearnaldi thanks for the bug report!

Indeed, looks like we left a comment to ["ignore resource routes"](https://github.com/remix-run/remix/blob/dev/packages/remix-dev/vite/plugin.ts#L641-L647) when removing server exports for the browser.
But "ignore" in that context should mean "return an empty module" not "return the original module" 🤦

Should be an easy fix!

**Comment by mikearnaldi** [comment]
> @mikearnaldi thanks for the bug report!
> 
> Indeed, looks like we left a comment to ["ignore resource routes"](https://github.com/remix-run/remix/blob/dev/packages/remix-dev/vite/plugin.ts#L641-L647) when removing server exports for the browser. But "ignore" in that context should mean "return an empty module" not "return the original module" 🤦
> 
> Should be an easy fix!

Let me know if I can help in any way, congrats on shipping vite integration, for Effect's integration it's a game-changer!

**Comment by pcattori** [comment]
Fixed by #7828

**Comment by mmachatschek** [comment]
Can confirm, the build errors are gone with that fix 👍
